The code was written by Jens Hansen working for the Ravi Iyengar Lab.<br>
The copyright holder for this preprint is the author/funder. It is made available under a creative commons 4.0 International license (CC BY 4.0).
Please acknowledge the MBC Ontology in your publications by citing the following reference:<br/>
<br/>
<b>Jens Hansen, Ravi Iyengar. MBC PathNet: integration and visualization of networks connecting functionally related pathways predicted from transcriptomic and proteomic datasets. Bioinform Adv. 2025 Aug 18;5(1). PMID: 40917650 </b>
___________________________________________________________________________________________________________
MBCO PathNet<br/>
MBC PathNet is made available under an Apache 2.0 license.<br/>
<br/>
Please acknowledge our application in your publications by citing the following reference:<br/>
<br/>
<b>MBCO PathNet: Integration and visualization of networks connecting functionally related pathways predicted from transcriptomic and proteomic datasets.
doi: https://doi.org/10.1101/2025.02.19.638900<b/>
___________________________________________________________________________________________________________
The Molecular Biology of the Cell Ontology (MBCO) datasets are made available under a Attribution-NonCommercial 2.0 Generic (CC BY-NC 2.0) International license.
___________________________________________________________________________________________________________
If you use any of the MBCO datasets or programming scripts described above, please acknowledge the MBC Ontology in your publications by citing the following reference:<br/>
Jens Hansen, David Meretzky, Simeneh Woldesenbet, Gustavo Stolovitzky, Ravi Iyengar: A flexible ontology for inference of emergent whole cell function from relationships between subcellular processes Sci Rep. 2017 Dec 18;7(1):17689. doi: 10.1038/s41598-017-16627-4.
___________________________________________________________________________________________________________
License for example experimental datasets used in the MBCO windows application

Neurite outgrowth dataset:<br/>
This dataset is made available under a Creative Commons Attribution 4.0 International (CC BY 4.0).<br>
Please cite the following reference if you use this data:<br>
Whole cell response to receptor stimulation involves many deep and distributed subcellular biochemical processes. Hansen J, Siddiq MM, Yadaw AS, Tolentino RE, Rabinovich V, Jayaraman G, Jain MR, Liu T, Li H, Xiong Y, Goldfarb J, Iyengar R. J Biol Chem. 2022 Aug 1:102325. doi: 10.1016/j.jbc.2022.102325. PMID: 35926710<br>
<br>
KPMP reference atlas datasets:<br/>
These datasets are made available under a Creative Commons Attribution 4.0 International (CC BY 4.0) and also can be downloaded from https://www.kpmp.org/doi-collection.<br/>
Please cite the following manuscripts if you use any of the datasets integrated in our KPMP reference atlas paper:<br/>
1) A reference tissue atlas for the human kidney. Hansen J, Sealfon R, Menon R, Eadon MT, Lake BB, Steck B, Anjani K, Parikh S, Sigdel TK, Zhang G, Velickovic D, Barwinska D, Alexandrov T, Dobi D, Rashmi P, Otto EA, Rivera M, Rose MP, Anderton CR, Shapiro JP, Pamreddy A, Winfree S, Xiong Y, He Y, de Boer IH, Hodgin JB, Barisoni L, Naik AS, Sharma K, Sarwal MM, Zhang K, Himmelfarb J, Rovin B, El-Achkar TM, Laszik Z, He JC, Dagher PC, Valerius MT, Jain S, Satlin LM, Troyanskaya OG, Kretzler M, Iyengar R, Azeloglu EU; Kidney Precision Medicine Project. Sci Adv. 2022 Jun 10;8(23):eabn4965. doi: 10.1126/sciadv.abn4965. Epub 2022 Jun 8. PMID: 35675394  (Laser microdissected RNAseq/Proteomics, Near Single Cell Proteomics)<br/>
2) A single-nucleus RNA-sequencing pipeline to decipher the molecular anatomy and pathophysiology of human kidneys. Lake BB, Chen S, Hoshi M, Plongthongkum N, Salamon D, Knoten A, Vijayan A, Venkatesh R, Kim EH, Gao D, Gaut J, Zhang K, Jain S. Nat Commun. 2019 Jun 27;10(1):2832. doi: 10.1038/s41467-019-10861-2. PMID: 31249312 (Single Nucleus RNAseq)<br/>
3) Single cell transcriptomics identifies focal segmental glomerulosclerosis remission endothelial biomarker. Menon R, Otto EA, Hoover P, Eddy S, Mariani L, Godfrey B, Berthier CC, Eichinger F, Subramanian L, Harder J, Ju W, Nair V, Larkina M, Naik AS, Luo J, Jain S, Sealfon R, Troyanskaya O, Hacohen N, Hodgin JB, Kretzler M, Kpmp KPMP; Nephrotic Syndrome Study Network (NEPTUNE). JCI Insight. 2020 Mar 26;5(6):e133267. doi: 10.1172/jci.insight.133267. PMID: 32107344 (Single Cell RNAseq)<br/>

___________________________________________________________________________________________________________

License for additional MBCO datasets ("Mbco_na_glucsose_tm_transport_human" in the MBCO application)<br/>
The additional MBCO datasets "Metabolic_energy_generation_pathways_human.txt", "NaAndGluTMTransport_scpGeneAssociations.txt", "NaAndGluTMTransport_scpHierarchy.txt" and "Sphingolipid_metabolism.txt" were generated by Jens hansen working for the Ravi Iyengar Lab and the Kidney Precision Medicine Project (KPMP). The "NaAndGluTMTransport-" files are part of the application where they can be accessed under the name "Mbco_na_glucsose_tm_transport_human".
The copyright holder for these datasets is the author/funder.
They are made available under a Creative Commons Attribution 4.0 Unported License (CC BY 4.0) International liscence.

To generate the dataset "NaAndGluTMTransport_scpGeneAssociations.txt" we started by extracting sodium and glucose transport related genes from the MBCO datasets, Wikipedia articles, peer-reviewed articles and Gene Ontology Biological Processes and Molecular Functions. See the methods section of Hansen, Sealfon, Menon et al. for details. Gene Ontology datasets were made available under a the Creative Commons Attribution 4.0 Unported License (http://geneontology.org/docs/go-citation-policy/#:~:text=Gene%20Ontology%20Consortium%20data%20and%20data%20products%20are,Creative%20Commons%20FAQ%20or%20contact%20the%20GO%20Helpdesk.), as on 2022 March 25..

Please cite our KPMP manuscript when using these datasets:<br/>
A reference tissue atlas for the human kidney. Hansen J, Sealfon R, Menon R, Eadon MT, Lake BB, Steck B, Anjani K, Parikh S, Sigdel TK, Zhang G, Velickovic D, Barwinska D, Alexandrov T, Dobi D, Rashmi P, Otto EA, Rivera M, Rose MP, Anderton CR, Shapiro JP, Pamreddy A, Winfree S, Xiong Y, He Y, de Boer IH, Hodgin JB, Barisoni L, Naik AS, Sharma K, Sarwal MM, Zhang K, Himmelfarb J, Rovin B, El-Achkar TM, Laszik Z, He JC, Dagher PC, Valerius MT, Jain S, Satlin LM, Troyanskaya OG, Kretzler M, Iyengar R, Azeloglu EU; Kidney Precision Medicine Project. Sci Adv. 2022 Jun 10;8(23):eabn4965. doi: 10.1126/sciadv.abn4965. Epub 2022 Jun 8. PMID: 35675394<br/>

Please visit one of the following websites to download the R and C# code that uses the additional MBCO datasets for the analysis of KPMP reference tissue data:<br/> 
https://github.com/KPMP/Reference-Tissue-Cell-Atlas-Manuscript-2022<br/>
KPMP containers deposited at Docker Hub<br/>
Windows Container for C# program: https://hub.docker.com/r/iyengarlab/kpmp-cs-apps<br/>
Linux Container for R and Python programs: https://hub.docker.com/r/iyengarlab/kpmp-rp-apps<br/>
README for Using KPMP Docker Containers: https://github.com/DToxS/KPMP-Contr-Apps<br/>
